sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_file_attribute.hpp

```diff
@@ -2,218 +2,82 @@
-      }
-      else if (fileAttributesString == "NoScrubData")
-      {
-        result = Models::FileAttributes::NoScrubData;
-      }
-
-      return result;
-    }
-
-    inline std::string FileAttributesToString(const Models::FileAttributes& val)
-    {
-      Models::FileAttributes value_list[] = {
-          Models::FileAttributes::ReadOnly,
-          Models::FileAttributes::Hidden,
-          Models::FileAttributes::System,
-          Models::FileAttributes::None,
-          Models::FileAttributes::Directory,
-          Models::FileAttributes::Archive,
-          Models::FileAttributes::Temporary,
-          Models::FileAttributes::Offline,
-          Models::FileAttributes::NotContentIndexed,
-          Models::FileAttributes::NoScrubData,
-      };
-      const char* string_list[] = {
-          "ReadOnly",
-          "Hidden",
-          "System",
-          "None",
-          "Directory",
-          "Archive",
-          "Temporary",
-          "Offline",
-          "NotContentIndexed",
-          "NoScrubData",
-      };
-      std::string result;
-      for (std::size_t i = 0; i < sizeof(value_list) / sizeof(Models::ListSharesIncludeType); ++i)
-      {
-        if ((val & value_list[i]) == value_list[i])
-        {
-          if (!result.empty())
-          {
-            result += "|";
-          }
-          result += string_list[i];
-        }
-      }
-      return result;
-    }
-
-    inline Models::FileAttributes FileAttributesListFromString(
-        const std::string& fileAttributesString)
-    {
-      Models::FileAttributes results = static_cast<Models::FileAttributes>(0);
-
-      std::string::const_iterator cur = fileAttributesString.begin();
-
-      auto getSubstrTillDelimiter
-          = [](char delimiter, const std::string& string, std::string::const_iterator& cur) {
-              auto begin = cur;
-              auto end = std::find(cur, string.end(), delimiter);
-              cur = end;
-              if (cur != string.end())
-              {
-                ++cur;
-              }
-              return std::string(begin, end);
-            };
-
-      while (cur != fileAttributesString.end())
-      {
-        std::string attribute = getSubstrTillDelimiter('|', fileAttributesString, cur);
-
-        if (!attribute.empty())
-        {
-          results |= FileAttributesFromString(attribute);
-        }
-      }
-
-      return results;
-    }
-  } // namespace Details
-
-}}}} // namespace Azure::Storage::Files::Shares
\ No newline at end of file
+namespace Azure { namespace Storage { namespace Files { namespace Shares { namespace Models {
+  class FileAttributes {
+  public:
+    FileAttributes() = default;
+    explicit FileAttributes(std::string value);
+    bool operator==(const FileAttributes& other) const;
+    bool operator!=(const FileAttributes& other) const;
+    FileAttributes operator|(FileAttributes rhs);
+    FileAttributes operator&(FileAttributes rhs);
+    const std::string& Get();
```

Comment:
Should this return a `std::string` or `std::vector<std::string>`

Reply:
Or do we need both functions, one returning `std::string`, one returning `std::vector<std::string>`.

Reply:
I do think we need the `std::string` return value which is used when sending the request. I don't know when getting the vector would be in use.

Reply:
What if a new symbol is added in the future, but our client sdk hasn't supported it yet. Now the user needs to be able to get the new symbol, right? This is the original intention of extensible enum. How is the user gonna do that?

Reply:
Makes sense.